src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsTerminal/Helpers/SettingsManager.cs

```diff
@@ -0,0 +1,144 @@
+    private readonly ToggleSetting _showHiddenProfiles = new(ShowHiddenProfiles, Resources.show_hidden_profiles, Resources.show_hidden_profiles, false);
+    private readonly ToggleSetting _openNewTab = new(OpenNewTab, Resources.open_new_tab, Resources.open_new_tab, false);
+    private readonly ToggleSetting _openQuake = new(OpenQuake, Resources.open_quake, Resources.open_quake_description, false);
+
+    private readonly Dictionary<string, object> settingsDict = new();
+
+    public const string ShowHiddenProfiles = nameof(ShowHiddenProfiles);
+    public const string OpenNewTab = nameof(OpenNewTab);
+    public const string OpenQuake = nameof(OpenQuake);
+
+    private static readonly JsonSerializerOptions _serializerOptions = new()
+    {
+        WriteIndented = true,
+        Converters = { new JsonStringEnumConverter() },
+    };
+
+    internal static string SettingsJsonPath()
+    {
+        // Get the path to our exe
+        var path = System.Reflection.Assembly.GetExecutingAssembly().Location;
+
+        // Get the directory of the exe
+        var directory = Path.GetDirectoryName(path) ?? string.Empty;
+
+        // now, the state is just next to the exe
+        return Path.Combine(directory, "state.json");
+    }
+
+    public SettingsManager()
+    {
+        _filePath = SettingsJsonPath();
+
+        _settings.Add(_showHiddenProfiles);
+        _settings.Add(_openNewTab);
+        _settings.Add(_openQuake);
+
+        settingsDict[ShowHiddenProfiles] = _showHiddenProfiles.ToDictionary();
+        settingsDict[OpenNewTab] = _openNewTab.ToDictionary();
+        settingsDict[OpenQuake] = _openQuake.ToDictionary();
+
+        // Load settings from file upon initialization
+        LoadSettings();
+    }
+
+    public Settings GetSettings()
+    {
+        return _settings;
+    }
+
+    public void SaveSettings()
+    {
+        try
+        {
+            settingsDict[ShowHiddenProfiles] = _showHiddenProfiles.ToDictionary();
+            settingsDict[OpenNewTab] = _openNewTab.ToDictionary();
+            settingsDict[OpenQuake] = _openQuake.ToDictionary();
+
+            // Serialize the main dictionary to JSON and save it to the file
+            var settingsJson = JsonSerializer.Serialize(settingsDict, _serializerOptions);
+            File.WriteAllText(_filePath, settingsJson);
+        }
+        catch (Exception ex)
+        {
+            ExtensionHost.LogMessage(new LogMessage() { Message = ex.ToString() });
+        }
+    }
+
+    public void LoadSettings()
+    {
+        if (!File.Exists(_filePath))
+        {
+            ExtensionHost.LogMessage(new LogMessage() { Message = "The provided settings file does not exist" });
+            return;
+        }
+
+        try
+        {
+            // Read the JSON content from the file
+            var jsonContent = File.ReadAllText(_filePath);
+
+            // Parse the JSON content into a JsonObject
+            if (JsonNode.Parse(jsonContent) is JsonObject savedSettings)
```

Comment:
oh god we should _not_ be asking extension authors to do this. Did I forget a `ToPayload` or `ToJson` on `Settings`? oh dear I did. 

One moment please...

Reply:
Take a look at `dev/migrie/more-settings-for-jordi` and if you like that, merge it in

Reply:
Those changes are really nice. Makes adding additional settings really straightforward too.